frontend/components/containers/comments/CommentList.vue

```diff
@@ -0,0 +1,143 @@
+    @input="updateSelectedComments"
+  >
+    <template v-slot:item.created_at="{ item }">
+      <span>{{ item.created_at | dateParse('YYYY-MM-DDTHH:mm:ss') | dateFormat('YYYY-MM-DD HH:mm') }}</span>
+    </template>
+    <template v-slot:top>
+      <v-text-field
+        v-model="search"
+        prepend-inner-icon="search"
+        :label="$t('generic.search')"
+        single-line
+        hide-details
+        filled
+      />
+    </template>
+    <template v-slot:item.action="{ item }">
+      <v-btn
+        small
+        color="primary text-capitalize"
+        @click="toLabeling(item)"
+      >
+        {{ $t('comments.commentView') }}
+      </v-btn>
+    </template>
+  </v-data-table>
+</template>
+
+<script>
+import Vue from 'vue'
+import { mapActions, mapGetters, mapState, mapMutations } from 'vuex'
+import VueFilterDateFormat from '@vuejs-community/vue-filter-date-format'
+import VueFilterDateParse from '@vuejs-community/vue-filter-date-parse'
+Vue.use(VueFilterDateFormat)
+Vue.use(VueFilterDateParse)
+
+export default {
+  async fetch() {
+    await this.getDocumentList({
+      projectId: this.$route.params.id,
+      ...this.$route.query
+    })
+    await this.getCommentListComplete({
+      projectId: this.$route.params.id
+    })
+  },
+  data() {
+    return {
+      search: this.$route.query.q,
+      options: {},
+      headers: [
+        {
+          text: 'Created At',
+          align: 'left',
+          value: 'created_at'
+        },
+        {
+          text: 'Document',
+          value: 'document_text'
+        },
+        {
+          text: 'Username',
+          value: 'username'
+        },
+        {
+          text: 'Text',
+          value: 'text'
+        },
+        {
+          text: this.$t('dataset.action'),
+          align: 'left',
+          value: 'action',
+          sortable: false
+        }
+      ]
+    }
+  },
+  computed: {
+    ...mapState('documents', ['items', 'comments', 'loading', 'selectedComments']),
+    ...mapGetters('projects', ['getLink'])
+  },
+  watch: {
+    '$route.query': '$fetch',
+    options: {
+      handler(newvalue, oldvalue) {
+        this.$router.push({
+          query: {
+            limit: this.options.itemsPerPage,
+            offset: (this.options.page - 1) * this.options.itemsPerPage,
+            q: this.search
+          }
+        })
+      },
+      deep: true
+    },
+    search() {
+      this.$router.push({
+        query: {
+          limit: this.options.itemsPerPage,
+          offset: 0,
+          q: this.search
+        }
+      })
+      this.options.page = 1
+    }
+  },
+  methods: {
+    ...mapActions('documents', ['getCommentListComplete', 'getDocumentList']),
+    ...mapMutations('documents', ['updateSelectedComments']),
+
+    toLabeling(comment) {
+      const index = this.items.findIndex(item => item.id === comment.document)
+      const offset = (this.options.page - 1) * this.options.itemsPerPage
+      const page = offset + index + 1
```

Comment:
Document page number calculation cannot be based on the comment list. I'm getting some 404's with this. It should be possible to directly use the document ID, that is provided in the comment object.

Reply:
This method is basically copied from the DocumentList. The items are all documents. The only thing I changed is that I get the document id from the clicked comment object instead of the clicked document like its done in the DocumentList.vue. I try to reproduce the 404 error.

Reply:
Ok, I could reproduce the 404. If you switch on the next page of comments and try to show a comment on that page it gives a 404.